### PR TITLE
External Network: connection controller

### PR DESCRIPTION
**Pull request by cheina97**
# Description

This PR introduces a container that implements the generic features, like connection check and firewall.

This PR focuses only on Connection resources and connection checking, migrating the ConnChecker already used in the current network.

### CODE REVIEW

**Review by fra98 — APPROVED** [review]

File: pkg/gateway/connection/connections-controller.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+	remoteIP, err := common.GetRemoteInterfaceIP(r.Options.Mode)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("unable to get the remote interface IP: %w", err)
+	}
+
+	go r.ConnChecker.AddAndRunSender(
+		ctx, r.Options.RemoteClusterID, remoteIP,
+		ForgeUpdateConnectionCallback(ctx, r.Client, req),
+	)
```
Comment (aleoli) [inline pkg/gateway/connection/connections-controller.go:91]: this is not guaranteed to be thread safe; the check if it is running and the set of the running one are safe

File: pkg/gateway/tunnel/wireguard/dns.go [hunk 2]
```diff
@@ -106,10 +106,10 @@ func NewDNSEventHandler(cl client.Client, opts *Options) handler.EventHandler {
 	return handler.EnqueueRequestsFromMapFunc(
 		func(ctx context.Context, _ client.Object) []reconcile.Request {
 			labelSet := labels.Set{
-				string(LabelsMode):             string(opts.Mode),
-				string(consts.RemoteClusterID): opts.RemoteClusterID,
+				string(LabelsMode):             string(opts.GwOptions.Mode),
+				string(consts.RemoteClusterID): string(opts.GwOptions.Mode),
```
Comment (aleoli) [inline pkg/gateway/tunnel/wireguard/dns.go:110]:
```suggestion
				string(consts.RemoteClusterID): string(opts.GwOptions.RemoteClusterID),
```

**Review by aleoli — APPROVED** [review]

**Review by aleoli — APPROVED** [review]

**Pull request merged**